warp/_src/context.py

```
@@ -8633,30 +8749,71 @@ def add_transform_type_stub(cls, label):
+    # =========================================================================
+    # Step 4: Generate merged stubs for function conflicts
+    # =========================================================================
+    def format_signature(func) -> str | None:
+        """Format function signature for stubs, cleaning up namespace prefixes."""
+        try:
+            sig = str(inspect.signature(func))
+            sig = sig.replace("warp.", "").replace("typing.", "")
+            sig = re.sub(r"<class '([^']+)'>", r"\1", sig)
+            # Replace bool with _builtins.bool to avoid shadowing by warp._src.types.bool
+            sig = re.sub(r"\bbool\b", "_builtins.bool", sig)
+            return sig
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)